src/test/java/org/minions/devfund/chelo/SquarelotronTest.java

```diff
@@ -0,0 +1,227 @@
+    public void testNotUpsideDownFlip() {
+        final int size = 3;
+        final int ring = 1;
+        final int[][] expectedArray = new int[][]{
+                {7, 8, 7},
+                {4, 5, 6},
+                {1, 2, 3}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        assertNotEquals(expectedArray, squarelotron.upsideDownFlip(ring).squarelotron);
+    }
+
+    /**
+     * A test to validate the diagonal flip.
+     */
+    @Test
+    public void testDiagonalFlip() {
+        final int size = 3;
+        final int ring = 1;
+        final int[][] expectedArray = new int[][]{
+                {1, 4, 7},
+                {2, 5, 8},
+                {3, 6, 9}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        assertEquals(expectedArray, squarelotron.mainDiagonalFlip(ring).squarelotron);
+    }
+
+    /**
+     * A test to validate the diagonal flip.
+     */
+    @Test
+    public void testNotDiagonalFlip() {
+        final int size = 3;
+        final int ring = 1;
+        final int[][] expectedArray = new int[][]{
+                {7, 4, 1},
+                {8, 5, 2},
+                {9, 6, 3}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        assertNotEquals(expectedArray, squarelotron.mainDiagonalFlip(ring).squarelotron);
+    }
+
+    /**
+     * A test to validate the rotate right.
+     */
+    @Test
+    public void testRotateRight1() {
+        final int size = 3;
+        final int rotate = 1;
+        final int[][] expectedArray = new int[][]{
+                {7, 4, 1},
+                {8, 5, 2},
+                {9, 6, 3}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        squarelotron.rotateRight(rotate);
+        assertEquals(expectedArray, squarelotron.squarelotron);
+    }
+
+    /**
+     * A test to validate the rotate right two.
+     */
+    @Test
+    public void testRotateRight2() {
+        final int size = 3;
+        final int rotate = 2;
+        final int[][] expectedArray = new int[][]{
+                {9, 8, 7},
+                {6, 5, 4},
+                {3, 2, 1}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        squarelotron.rotateRight(rotate);
+        assertEquals(expectedArray, squarelotron.squarelotron);
+    }
+
+    /**
+     * A test to validate the rotate right six.
+     */
+    @Test
+    public void testRotateRight6() {
+        final int size = 4;
+        final int rotate = 6;
+        final int[][] expectedArray = new int[][]{
+                {16, 15, 14, 13},
+                {12, 11, 10, 9},
+                {8, 7, 6, 5},
+                {4, 3, 2, 1}};
+
+        Squarelotron squarelotron = new Squarelotron(size);
+        squarelotron.rotateRight(rotate);
+        assertEquals(expectedArray, squarelotron.squarelotron);
+    }
+
+    /**
+     * A test to validate the rotate right three.
+     */
+    @Test
+    public void testRotateRight1dimensionThree() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "testRotateRight1" on line 163. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)